cmd/kpod/parse.go

```diff
@@ -0,0 +1,872 @@
+// most of these validate and parse functions have been taken from projectatomic/docker
+// and modified for cri-o
+package main
```

Comment:
Is this code that we could drop in some common project that kpod, docker, and maybe others could get to?  I'm just concerned that one of the functions here has a noted issue in Buildah and once this is merged we'll have two places to fix if an error pops.

Reply:
We could move it up to cri-o/parsers subdir and then buildah/kpod could vendor it in.  (If kpod gets split out.